src/zjs_promise.c

```diff
@@ -124,46 +128,64 @@ void zjs_make_promise(jerry_value_t obj, zjs_post_promise_func post,
 
 void zjs_fulfill_promise(jerry_value_t obj, jerry_value_t argv[], uint32_t argc)
 {
-    struct promise* handle;
+    struct promise* handle = NULL;
     jerry_value_t promise_obj = zjs_get_property(obj, "promise");
 
+    if (!jerry_value_is_object(promise_obj)) {
+        ERR_PRINT("'promise' not found in object %lu\n", obj);
+        return;
+    }
+
     jerry_get_object_native_handle(promise_obj, (uintptr_t*)&handle);
 
-    // Put *something* here in case it never gets registered
-    if (!handle->then_set) {
-        handle->then = jerry_create_external_function(null_function);
-    }
+    if (handle) {
+        // Put *something* here in case it never gets registered
+        if (!handle->then_set) {
+            handle->then = jerry_create_external_function(null_function);
+        }
 
-    handle->then_id = zjs_add_callback_once(handle->then,
-                                            obj,
-                                            handle,
-                                            post_promise);
+        handle->then_id = zjs_add_callback_once(handle->then,
+                                                handle->this,
+                                                handle,
+                                                post_promise);
 
-    zjs_signal_callback(handle->then_id, argv, argc * sizeof(jerry_value_t));
+        zjs_signal_callback(handle->then_id, argv, argc * sizeof(jerry_value_t));
 
-    DBG_PRINT("fulfilling promise, obj=%lu, then_id=%lu, argv=%p, nargs=%lu\n",
-              obj, handle->then_id, argv, argc);
+        DBG_PRINT("fulfilling promise, obj=%lu, then_id=%lu, argv=%p, nargs=%lu\n",
+                obj, handle->then_id, argv, argc);
```

Comment:
Minor, but either obj should line up with " or be 4-chars in, I think we allow either.